File: ios/KeyCommonNative.swift
//
//  KeyCommonNative.swift
//  onLineUp
//
//  Created by lam.nguyen5 on 11/18/21.
//

import Foundation

class KeyCommonNative {
  public static let COMMON_NATIVE_EMITTING = "COMMON_NATIVE_EMITTING";
  
  // oneShotStorage keys
  public static let DATA_FROM_NOTIFICATION = "DATA_FROM_NOTIFICATION";
  public static let DATA_FROM_DEEPLINK = "DATA_FROM_DEEPLINK";
  
  public static func getAllKey() -> [String: String] {
    return [
      COMMON_NATIVE_EMITTING: COMMON_NATIVE_EMITTING,
      DATA_FROM_NOTIFICATION: DATA_FROM_NOTIFICATION,
      DATA_FROM_DEEPLINK: DATA_FROM_DEEPLINK
    ]
  }
}

File: android/app/src/main/java/com/onlineup/core/nativepackage/coreapi/constant/KeyCommonNative.java
package com.onlineup.core.nativepackage.coreapi.constant;

import java.util.HashMap;
import java.util.Map;

public class KeyCommonNative {
    public static String COMMON_NATIVE_EMITTING = "COMMON_NATIVE_EMITTING";

    // oneShotStorage keys
    public static String DATA_FROM_NOTIFICATION = "DATA_FROM_NOTIFICATION";
    public static String DATA_FROM_DEEPLINK = "DATA_FROM_DEEPLINK";

    public static Map<String, Object> getAllKeys() {
        Map<String, Object> map = new HashMap<>();
        map.put(COMMON_NATIVE_EMITTING, COMMON_NATIVE_EMITTING);
        map.put(DATA_FROM_NOTIFICATION, DATA_FROM_NOTIFICATION);
        map.put(DATA_FROM_DEEPLINK, DATA_FROM_DEEPLINK);

        return map;
    }

}
